src/StatusQ/Buttons/StatusIconButton.qml

```diff
@@ -0,0 +1,82 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusIconButton
+
+    property color iconNormalColor: "transparent"
+    property color iconDisabledColor: "transparent"
+
+    property color backgroundNormalColor: "transparent"
+    property color backgroundHoverColor: "transparent"
+    property color backgroundDisabledColor: "transparent"
+    property alias iconName: icon.icon
+
+    property bool showBackground: false
+
+    property bool loading: false
+
+    signal pressed(var mouse)
+
+
+
+    /// Implementation
+
+    implicitWidth: 44// by design
+    implicitHeight: 44// by design
+    radius: showBackground ? width / 2
+                           : 8
+
+    color: {
+        if (!showBackground) {
+            return sensor.containsMouse ? backgroundHoverColor
+                                        : "transparent"
+        }
+
+        if (statusIconButton.enabled) {
+            return sensor.containsMouse ? backgroundHoverColor
+                                        : backgroundNormalColor
+        } else {
+            return backgroundDisabledColor
+        }
+    }
+    MouseArea {
+        id: sensor
+
+        anchors.fill: parent
+        cursorShape: Qt.PointingHandCursor
+        hoverEnabled: !loading
+
+
+        StatusIcon {
+            id: icon
+            anchors.centerIn: parent
+            visible: !loading
+
+            width: statusIconButton.width -  20
+            height: statusIconButton.height - 20
+
+            color: {
+                if (statusIconButton.enabled) {
+                    return iconNormalColor
+                } else {
+                    return iconDisabledColor
+                }
+            }
+        } // Icon
+        StatusLoadingIndicator {
+            anchors.centerIn: parent
+            visible: loading
+            color: {
+                if (statusIconButton.enabled) {
+                    return iconNormalColor
+                } else {
+                    return iconDisabledColor
+                }
+            }
+        } // Indicator
+    } // Sensor
+} // Rectangle
```

Comment:
Another thing to keep in mind here:

This component is currently called `StatusRoundButton` (analogue to QML's `RoundButton`), while `StatusIconButton` is a different component.

If we aren't careful, this is going  to break many places in the app.

How do you feel about sticking with `StatusRoundButton` as a name for now?

Reply:
No problem